core/resolve/src/mill/resolve/ResolveCore.scala

```diff
@@ -448,15 +448,15 @@ private object ResolveCore {
 
     val namedTasks = Reflect
       .reflect(cls, classOf[Task.Named[?]], namePred, noParams = true, cache.getMethods)
-      .map { m =>
-        Resolved.NamedTask(Segments.labels(cache.decode(m.getName))) ->
-          None
+      .collect {
+        case m if !classOf[Task.Command[?]].isAssignableFrom(m.getReturnType) =>
+          Resolved.NamedTask(Segments.labels(cache.decode(m.getName))) -> None
```

Comment:
`collect` may silently drop some items, while the previous `map` was complete. I think we want a default case to catch that case or a comment explaining, why this isn't needed.